### PR TITLE
NEW Add font-icon support for site tree

### PR DESCRIPTION
**Pull request by sachajudd**
Resolves: https://github.com/silverstripe/silverstripe-admin/issues/574
Relies on: https://github.com/silverstripe/silverstripe-admin/pull/803

### CODE REVIEW

File: code/Model/SiteTree.php [hunk 1]
```diff
@@ -277,6 +277,13 @@ class SiteTree extends DataObject implements PermissionProvider, i18nEntityProvi
      */
     private static $icon = null;
 
+    /**
+     * Class attached to page icons in the CMS page tree. Also supports font-icon set.
+     * @config
+     * @var string
+     */
+    private static $icon_class = 'font-icon-page';
```
Comment (maxime-rainville) [inline code/Model/SiteTree.php]: Can we document this somewhere?

File: code/Model/SiteTree.php [hunk 2]
```diff
@@ -2747,7 +2754,8 @@ public function getTreeTitle()
         $children = $this->creatableChildren();
         $flags = $this->getStatusFlags();
         $treeTitle = sprintf(
-            "<span class=\"jstree-pageicon page-icon class-%s\"></span><span class=\"item\" data-allowedchildren=\"%s\">%s</span>",
+            "<span class=\"jstree-pageicon page-icon %s class-%s\"></span><span class=\"item\" data-allowedchildren=\"%s\">%s</span>",
```
Comment (maxime-rainville) [inline code/Model/SiteTree.php]: Not a big fan of writing HTML in PHP. But it's no worst than it was.

File: code/Model/SiteTree.php (hunk 1 above)
Comment (sachajudd) [reply code/Model/SiteTree.php]: Yeah sure thing, I'll create a PR later today 🙂

File: code/Model/SiteTree.php [hunk 3]
```diff
@@ -2747,7 +2754,8 @@ public function getTreeTitle()
...
+            $this->config()->get('icon') ? '' : $this->config()->get('icon_class'),
```
Comment (robbieaverill) [inline code/Model/SiteTree.php]: I feel like this deserves to be in its own method, e.g. `getIconClass()` or something, but it's not a blocker

**Pull request merged**